portfolio/src/main/java/com/google/sps/servlets/FormServlet.java

```diff
@@ -106,4 +126,39 @@ private String convertToJson(ArrayList<Comment> list) {
     String json = gson.toJson(list);
     return json;
   }
+
+  // Returns a URL that points to the uploaded file, or null if the user didn't upload a file
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    // Return null if user submits form without selecting a file (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Get the key of the file input
+    BlobKey blobKey = blobKeys.get(0);
+
+    // Return null if user submits form without selecting a file (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
```

Comment:
This runs differently in devserver vs production?

Reply:
I unfortunately do not know for sure because I haven't been able to use Blobstore in production yet, but getUploadFileUrl() was given to us during the tutorial and I figured I'd keep this part of the function because I can't test yet on the deployed website. I was thinking that the live server still stored some sort of information about a non-submitted blob but checking the meta data's size would reveal if an image was actually submitted.